### PR TITLE
[DistillationTrainer refactor] Add `_get_last_hidden_state` (unwired)

### PR DESCRIPTION
**Pull request by qgallouedec**
*Item 31 (Group E) — stacked on PR 30. Part of #6449.*

First step of the loss cutover to chunked JSD. Adds GRPO's `_get_last_hidden_state` (`grpo_trainer.py:1336`), which returns the backbone's `(B, logits_to_keep, H)` hidden states (skipping `lm_head`). The chunked JSD loss will project these to vocab per chunk, so only `(C, H)` is retained instead of `(C, V)` logits — the memory win.

Unwired here: nothing calls it yet (the current loss still runs through `generalized_jsd_loss`/`_get_teacher_logits`). It gets exercised once `compute_loss` is wired to the chunked path (item 33).

Aligned to GRPO verbatim, with one deferral:
- Uses `self.model_kwarg_keys` (added at item 25) to pass `logits_to_keep` only to models whose forward accepts it.
- The VLM image kwargs (`pixel_values`, `image_grid_thw`, …) default to `None`, so their branches are runtime no-ops until VLM lands (item 45) — same pattern as `_generate_single_turn`'s empty multimodal-fields loop.
- Backbone selection ported as the non-VLM path (`unwrapped_model.base_model`). GRPO's pre-5.0-transformers VLM fallback (`self._is_vlm` + version check) is restored at item 45 when `self._is_vlm` exists.

Import: `is_peft_model` from `accelerate.utils` (GRPO's source).

No test — add-unwired port, no GRPO analog; covered once wired (item 33). Verified: ruff clean; `test_distillation_trainer.py` (40) + `test_server_distillation_trainer.py` (12) green.

---

> [!NOTE]
> **Low Risk**
> Dead code path only: the new helper is not invoked, so current training behavior is unchanged.
> 
> **Overview**
> Adds **`_get_last_hidden_state`** to `DistillationTrainer` (ported from GRPO) as groundwork for switching distillation loss to **chunked JSD**. The helper runs the model **backbone** (not `lm_head`), optionally passes VLM kwargs and `logits_to_keep` when the forward signature supports it, unwraps PEFT models, and returns `(B, logits_to_keep, H)` hidden states for completion positions.
> 
> Also imports **`is_peft_model`** from `accelerate.utils`. **Nothing calls the new method yet**—training still uses `generalized_jsd_loss` / `_get_teacher_logits` until a follow-up wires `compute_loss` to the chunked path.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit c4c6433564131217c0c7a7e696b4539e7355141f. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/distillation/distillation_trainer.py [hunk 1]
```diff
@@ -762,6 +762,61 @@ def _generate(self, prompts: list):
 
         return prompt_ids, completion_ids
 
+    def _get_last_hidden_state(
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/experimental/distillation/distillation_trainer.py:772]: solved

**Review by cmpatino — APPROVED** [review]
